LogMonitor/src/LogMonitor/JsonProcessor.cpp

```diff
@@ -0,0 +1,284 @@
+
+            EventLogChannel channel(Utility::string_to_wstring(name), EventChannelLogLevel::Error);
+            // Set the level based on the levelString
+            if (channel.SetLevelByString(Utility::string_to_wstring(levelString))) {
+                // Successfully set the level
+            }
+            else {
+                logWriter.TraceError(
+                    Utility::FormatString(
+                        L"Invalid level string: %lu",
+                        Utility::string_to_wstring(levelString)
+                    ).c_str()
+                );
+            }
+            // Add the channel to the vector
+            channels->push_back(channel);
+        }
+        Attributes[JSON_TAG_CHANNELS] = channels;
+    }
+
+    std::shared_ptr<SourceEventLog> sourceEventLog = std::make_shared< SourceEventLog>();
+
+    //
+    // Fill the new EventLog source object, with its attributes
+    //
+    if (!SourceEventLog::Unwrap(Attributes, *sourceEventLog))
+    {
+        logWriter.TraceError(
+            L"Error parsing configuration file. "
+            L"Invalid EventLog source (it must have a non-empty 'channels')"
+        );
+    }
+
+    Sources.push_back(std::reinterpret_pointer_cast<LogSource>(std::move(sourceEventLog)));
+}
+
+// Function to handle File type
+void handleFile(
+    _In_ const boost::json::value& source,
+    _In_ AttributesMap& Attributes,
+    _Inout_ std::vector<std::shared_ptr<LogSource>>& Sources
+) {
+    std::string directory = source.as_object().at("directory").as_string().c_str();
+    std::string filter = source.as_object().at("filter").as_string().c_str();
+    bool includeSubdirs = source.at("includeSubdirectories").as_bool();
+    std::string customLogFormat = source.as_object().at("customLogFormat").as_string().c_str();
+
+    Attributes[JSON_TAG_DIRECTORY] = reinterpret_cast<void*>(new std::wstring(Utility::string_to_wstring(directory)));
+    Attributes[JSON_TAG_FILTER] = reinterpret_cast<void*>(new std::wstring(Utility::string_to_wstring(filter)));
+    Attributes[JSON_TAG_INCLUDE_SUBDIRECTORIES] = reinterpret_cast<void*>(
+        new std::wstring(includeSubdirs ? L"true" : L"false")
+        );
+    Attributes[JSON_TAG_CUSTOM_LOG_FORMAT] = reinterpret_cast<void*>(
+        new std::wstring(
+            Utility::string_to_wstring(customLogFormat)
+        )
+        );
+
+    std::shared_ptr<SourceFile> sourceFile = std::make_shared< SourceFile>();
+    //
+    // Fill the new FileLog source object, with its attributes
+    //
+    if (!SourceFile::Unwrap(Attributes, *sourceFile))
+    {
+        logWriter.TraceError(
+            L"Error parsing configuration file. "
+            L"Invalid File source (it must have a non-empty 'channels')"
+        );
+    }
+
+    Sources.push_back(std::reinterpret_pointer_cast<LogSource>(std::move(sourceFile)));
+}
+
+// Function to handle ETW type
+void handleETW(
```

Comment:
nit: could qualify this to `handleETWLog`

Reply:
This is resolved.